app/domain/operations/hbx_enrollments/drop_enrollment_members.rb

```diff
@@ -29,17 +30,19 @@ def drop_enrollment_members(params)
         terminate_date = Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y")
         dropped_enr_members = params[:options].select{|string| string.include?("terminate_member")}.values
         all_enr_members = params[:hbx_enrollment].hbx_enrollment_members
-        eligible_members = all_enr_members.reject{ |member| dropped_enr_members.include?(member.id.to_s) }
+        non_eligible_members = all_enr_members.select{ |member| dropped_enr_members.include?(member.id.to_s) }
+        new_effective_date = terminate_date + 1.day
+        optionals_params = params[:hbx_enrollment].is_ivl_by_kind? ? ivl_params(params[:hbx_enrollment], new_effective_date) : {}
+        clone_enrollment = Operations::HbxEnrollments::Clone.new.call({hbx_enrollment: params[:hbx_enrollment], effective_on: new_effective_date, options: optionals_params})
+        return clone_enrollment if clone_enrollment.failure?
 
-        reinstatement = Enrollments::Replicator::Reinstatement.new(params[:hbx_enrollment], terminate_date + 1.day, 0, eligible_members).build
-        reinstatement.save!
-
-        get_household_applied_aptc(reinstatement, terminate_date + 1.day) if params[:hbx_enrollment].applied_aptc_amount > 0
-
-        reinstatement.force_select_coverage! if reinstatement.may_reinstate_coverage?
-        reinstatement.begin_coverage! if reinstatement.may_begin_coverage? && reinstatement.effective_on <= TimeKeeper.date_of_record
-
-        params[:hbx_enrollment].update_attributes!(terminated_on: terminate_date)
+        new_enrollment = clone_enrollment.value!
+        remove_dropped_enr_members(new_enrollment, non_eligible_members)
```

Comment:
new_enrollment.hbx_enrollment_members.reject! {|mem| non_eligible_members.include?(mem.id)}

Reply:
We have already hbx_enrollment_members created in the clone operations which was suggested. So I had to delete the members as per the suggestion earlier.

Reply:
Don't save and delete member, after clone can you try above code snippet and see member are deleted cloned member object.